prime-router/src/main/kotlin/ActionLog.kt

```diff
@@ -93,7 +93,7 @@ interface ActionLogDetail {
     /**
      * The error code used to translate the error in the UI.
      */
-    val errorCode: String
+    val errorCode: ErrorCode
```

Comment:
I made this change because we should use the enum instead of its string presentation wherever possible for best coding practices.